docs/Datadog.Trace/README.md

```diff
@@ -0,0 +1,227 @@
+* `SpanTypes` associated with automatic instrumentation spans, such as `MongoDb` and `Redis` have been removed.  
+* `Tags` associated with automatic instrumentation spans, such as `AmqpCommand` and `CosmosDbContainer` have been removed.
+* `Tracer.Create()` has been removed. Use `Tracer.Configure()` instead.
+* `TracerSettings.AdoNetExcludedTypes` has been removed. Use `TracerSettings.Integrations` to disable automatic instrumentation for ADO.NET integrations in the same way as other integrations.
+* Various internal APIs that were not intended for public consumption have been removed. 
+
+In addition, some settings have been marked obsolete:
+
+* Environment variables `DD_TRACE_ANALYTICS_ENABLED`, `DD_TRACE_{0}_ANALYTICS_ENABLED`, and `DD_TRACE_{0}_ANALYTICS_SAMPLE_RATE` for controlling App Analytics are obsolete. App Analytics has been replaced with Tracing Without Limits. See [our documentation for details](https://docs.datadoghq.com/tracing/legacy_app_analytics/).
+* Setting `TracerSettings.AnalyticsEnabled` has similarly been marked obsolete.
+* Environment variable `DD_TRACE_LOG_PATH` is deprecated. Use `DD_TRACE_LOG_DIRECTORY` instead.
+
+### Introduction of ISpan and IScope
+
+In version 2.0 the public `Scope` and `Span` classes have been made internal. We now expose `IScope` and `ISpan` instead. The `Tracer` API has been updated accordingly. In most cases, this should not require any changes to your code, but if you are currently using explicit types (instead of inferred types), then you will need to replace usages of `Scope` with `IScope` and `Span` with `ISpan`:
+
+```csharp
+using Datadog.Trace;
+
+// No changes required here
+using (var scope = Tracer.Instance.StartActive("my-operation")
+{
+}
+
+// No Longer compiles (incorrect usage of Scope)
+using (Scope scope = Tracer.Instance.StartActive("my-operation")
+{
+}
+
+// Correct usage with explicit type
+using (IScope scope = Tracer.Instance.StartActive("my-operation")
+{
+}
+```
+
+### Simplification of the tracer interface
+
+In addition to returning `IScope`, the `Tracer.StartActive` method has been simplified to provide a `SpanCreationSettings`. In addition, we have removed the possibility to override the service name.
+
+
+### Incorrect integration names are ignored
+
+In version 2.0, any changes made to an `IntegrationSettings` object for an unknown `IntegrationId` will not be persisted. Instead, a warning will be logged describing the invalid access.
+
+```csharp
+using Datadog.Trace;
+
+var tracerSettings = TracerSettings.FromDefaultSources();
+
+// Accessing settings for an unknown integration will log a warning
+var settings = tracerSettings .Integrations["MyRandomIntegration"];
+
+// changes are not persisted
+settings.Enabled = false; 
+
+// isEnabled is null, not false
+bool? isEnabled = tracerSettings.Integrations["MyRandomIntegration"].Enabled;
+```
+
```

Comment:
I think it could help to subgroup the last 3 parts into a `### Automatic instrumentation changes` and make the previous h3's into h4's

Reply:
Implemented in 6d95cccfa92dc84cfcb1571a8608462735ff452a